### ISSUE TITLE
Unnecessary tile pop in when viewing a previously loaded area (memory caching bug)

### ISSUE THREAD

**Issue by csciguy8** [CesiumGS/cesium-unreal#1327]
1. Open the `cesium-unreal-samples` project

2. Open level `12_GoogleMapsTiles`

3. In the editor perspective window, you should see the Googleplex and surrounding area load

4. Move the camera 180 degrees so you are looking behind you

5. Move the camera back to the starting position

You will see the original tiles load again, faster, because they are cached on disk. This isn't the desired behavior though.

It is expected that you should **not** see any tiles load or pop in at all. The Cesium3DTileset's `Maximum Cached Bytes` is configured for 256MB, and all of these tiles only take up ~40MB in the disk cache (`cesium-request-cache.sqlite`). The tiles is in the original view should stay loaded in this scenario.

Thanks @kring for mentioning this last month.

**Comment by kring** [comment]
I noticed that changing the `Maximum Cached Bytes` property from 268435456 (256MiB) to 26843545600 (25600MiB) makes the problem go away. So the problem does seem to be that tiles are being unloaded when you look away due to the cache size being exceeded, but I'm not sure why.

**Comment by kring** [comment]
Turns out I was wrong about how the tile data size is measured for cache purposes. It's actually formed by adding up the number of bytes in all of the glTF buffers (`cesium.data.size()`) and images (`cesium.pixelData.size()`). That is _after_ things like Draco and JPG decompression, so it can be a lot larger than the size of the data downloaded over the network (even after it is gunzipped). A single 256x256 JPG ends up being 341KiB.

**Comment by kring** [comment]
Right, for that Googleplex scene, the actual size of the traversed tiles is about 300 MiB. I'm still trying to be 100% certain it's correct, but I think it is. In which case the default 256MiB max cache for Google Photorealistic 3D Tiles is simply too small.

**Comment by csciguy8** [comment]
Ok, if that's true, then my expectations were just wrong.

But this problem could lead to more work with this feature. For example, a new "auto" setting that determines a reasonable size based on your system memory or some other metrics. We could still have a fixed memory size for users that know exactly their device constraints.

**Comment by kring** [comment]
For comparison, the default in CesiumJS is 512MiB. It's not clear why ours is half that. Part of the reason may be that we're assuming there are extra copies kicking around in memory. cesium-native maintains a copy of all the tile data in the glTF data structure that isn't used at all for rendering. The game engines maintain the copy for rendering, of course, but may also maintain theirs own CPU-side copy as well.

**Comment by kring** [comment]
The Unreal "Memory" stat doesn't seem to work at all for showing our texture memory usage. If I set the cache size huge and fly around, the static mesh memory goes up steadily, as you'd expect. But the texture memory usage doesn't change at all.
<img width="815" alt="image" src="https://github.com/CesiumGS/cesium-unreal/assets/924374/54f76848-614d-4347-ae96-4870181d9bd5">

Perhaps our custom texture upload code is somehow evading UE's texture memory performance counter? 😬

**Comment by kring** [comment]
> Perhaps our custom texture upload code is somehow evading UE's texture memory performance counter?

Yes, this. If I force it to use the legacy texture upload path, the texture memory usage is much more realistic:
<img width="775" alt="image" src="https://github.com/CesiumGS/cesium-unreal/assets/924374/9ff0a45a-28c6-430a-9142-46d2c4a530fe">
